libs/src/sanityChecks/rolloverNodes.js

```diff
@@ -0,0 +1,55 @@
+const Utils = require('../utils')
+
+const TEN_SECONDS = 10000
+const MAX_TRIES = 3
+
+/** Check if the user's primary creator node is healthy */
+const checkPrimaryHealthy = async (libs, primary, tries) => {
+  const healthy = await Utils.isHealthy(primary)
+  if (healthy) return healthy
+  else {
+    if (tries === 0) {
+      return false
+    }
+    await Utils.wait(TEN_SECONDS)
+    return checkPrimaryHealthy(libs, primary, tries - 1)
+  }
+}
+
+/** Gets new endpoints from a user's secondaries */
+const getNewEndpoints = async (libs, secondaries) => {
+  for (const secondary of secondaries) {
+    const { status, userBlockNumber } = await libs.creatorNode.getSyncStatus(secondary)
+    if (status.blockNumber === userBlockNumber) {
```

Comment:
@raymondjacobson @SidSethi this checks that the creator node has synced to the latest transaction on chain. what if there's a case where it's not at the most recent block, but the next to last? do you think we should sort by blockNumber desc and pick the secondaries that way?

Reply:
if we pick secondaries that way, we could hypothetically select a secondary that is behind and cause the user to delegate that as their primary, which would lead to inconsistent data.
The block number here is the latest block number for which the user has changes

Reply:
@raymondjacobson so the two options i'm considering (and let me know if i'm forgetting / misunderstanding something) are 

1. we pick the creator node that's equal to the last known block number for that user. the pro is the user is never in an outdated state, but the con is if both secondaries are not caught up it will never roll over and we could potentially orphan that user without a primary?

2. we order the creator node endpoints in desc order of blocknumber. the con is we lose some state. the pro is that in the event that no backup is up to date we'd pick one that was closest.

Reply:
yeah. i think that's a fair summary. I'm super super super scared of #2 but I do agree with you that #1 is no bueno either. Maybe we can sync irl about this one

Reply:
@raymondjacobson discussed a good potential solution with draj - select "most uptodate" secondary in event of primary down instead of only "fully uptodate".

Reply:
@SidSethi @raymondjacobson yeah if we can special case then then i'm gucci here

Reply:
im not sure if i agree with this--maybe some-day we should do this, but i think it gets us into a more dangerous state. maybe let's sync up real quick

Reply:
as discussed, resolving this as it's is out scope for this task. should resolve afk